cfe_internal/enterprise/federation/federation.cf

```diff
@@ -33,13 +33,23 @@ bundle agent config
                        strcmp( "enabled", "$(data[remote_hubs][$(remotes)][target_state])" ),
                        # To ensure the remote we are pushing to actually needs the data (is a superhub)
                        strcmp( "superhub", "$(data[remote_hubs][$(remotes)][role])" ));
+
+      # Public keys of enabled pushing feeders need to be trusted (on a superhub)
+      # XXX: would it be okay to guard this with 'am_superhub::'???
```

Comment:
Hummm.
worst case (but I know that the agent looks deeper), following normal order, enabled is not set during first pass of vars, enabled gets defined during first pass of classes, second pass of vars data is loaded, second pass of classes am_superhub is defined, third pass of vars pubkey array if guarded with am_superhub is generated.

Yeah, I think that it would be fine to guard that with am_superhub, you should still resolve everything within the three passes.

In fact I think if data => readjson( path ) were moved outside of the enabled guard, and guarded with if => fileexists() then we might be able to reduce the number of passes required to reach full convergence.

Either way, yes, I think we can guard the generation of the pubkey array.